tests/System.Text.Utf8String.Tests/RandomTests.cs

```diff
@@ -631,6 +631,37 @@ public void IndexOfTests(int expected, string s, string substring)
         }
 
         [Theory]
+        [InlineData(0, "", "")]
+        [InlineData(0, "abc", "")]
```

Comment:
I tried it on .NET ( ```c# Console.WriteLine("abc".LastIndexOf(""));``` ) and it prints 2. I don't think it makes sense, but maybe we should be compatible? @stephentoub, @terrajobst, @joshfree, any opinions on this?

Reply:
Unless there's a good reason to not be compatible (eg significant perf impact) why would we not be intentionally compatible...

Reply:
Because I am not sure how returning 2 makes any sense here. :-)